ouroboros-network-framework/src/Ouroboros/Network/ConnectionHandler.hs

```diff
@@ -99,15 +104,27 @@ sduHandshakeTimeout = 10
 -- * 'HandleError'
 --                - the multiplexer thrown 'MuxError'.
 --
-data Handle (muxMode :: MuxMode) peerAddr versionData bytes m a b =
+data Handle (muxMode :: MuxMode) initiatorCtx responderCtx versionData bytes m a b =
     Handle {
         hMux            :: !(Mux muxMode m),
-        hMuxBundle      :: !(MuxBundle muxMode bytes m a b),
+        hMuxBundle      :: !(OuroborosBundle muxMode initiatorCtx responderCtx bytes m a b),
         hControlMessage :: !(TemperatureBundle (StrictTVar m ControlMessage)),
         hVersionData    :: !versionData
       }
 
 
+-- | 'Handle' used in P2P.
+--
+type HandleP2P muxMode peerAddr versionData bytes m a b =
+     Handle    muxMode (ExpandedInitiatorContext peerAddr m)
+                       (ResponderContext peerAddr)
+                       versionData bytes m a b
+
+type HandleNonP2P muxMode peerAddr versionData bytes m a b =
```

Comment:
^ find better names; these are used by `node-to-node` and `node-to-client`.

Reply:
Done.  I used `HandeWith{Expanded,Minimal}Ctx` and added haddocks how they are used.